### PR TITLE
Bugfix #1668923: WAV/AIFF metadata in RIFF tag

### PR DESCRIPTION
**Pull request by uklotzde**
Again, targeted at both master and 2.1. I'm still unsure about our merging strategy, until now it was from master to 2.1.

### CODE REVIEW

**Comment by uklotzde** [comment]
The CI build should reveal if the changes are compatible with older TagLib versions.

**Comment by uklotzde** [comment]
Force push after CI build stalled

**Comment by daschuer** [comment]
It works nice, thank you. 

AIFF tags are still not read, I have just noticed that they use a different format:
http://taglib.org/api/classTagLib_1_1RIFF_1_1AIFF_1_1File.html

Would you mind to add that as well?

**Comment by daschuer** [comment]
I have read the AIFF taglib side again, and it looks like it only supports ID3v2 Tags. 
Is this correct?

**Comment by daschuer** [comment]
.. or can we reed the Metadata via audio_properties?

**Comment by uklotzde** [comment]
I guess so. It doesn't look like TagLib supports any other kind of tags than 
ID3v2 for AIFF files.

The audio properties only contain information about the audio signal, no 
metadata.

**Comment by uklotzde** [comment]
The TagLib API has lots of oddities. I will add a small change (not really a 
fix) for AIFF files in a follow-up PR.

**Comment by uklotzde** [comment]
I added the small fix for AIFF files to this PR ;)

**Comment by daschuer** [comment]
there is a solution using the protected file member functions like that: 
```
            	  for(unsigned int i = 0; i < file.chunkCount(); i++) {
            	    const ByteVector chunkName = file.chunkName(i);
            	    if(chunkName == "NAME") {
            	    	const ByteVector chunkData file.chunkData(i);
            	        QString name = QString::fromLatin1(chunkData.data(), chunkData.size());
            	        pTrackMetadata->setTitle(name);
            	    } else if(chunkName == "AUTH") {
            	    	const ByteVector chunkData file.chunkData(i);
            	        QString name = QString::fromLatin1(chunkData.data(), chunkData.size());
            	        pTrackMetadata->setArtist(name);
            	    } else if(chunkName == "ANNO") {
            	    	const ByteVector chunkData file.chunkData(i);
            	        QString name = QString::fromLatin1(chunkData.data(), chunkData.size());
            	        pTrackMetadata->setComment(name);
            	    }
            	}
```

How stable are protected functions in a public API?

**Comment by uklotzde** [comment]
Essentially that means using a patched version of TagLib or applying some 
dirty tricks to get hold of these functions. No, I don't want to do that. An 
official feature request for extending TagLib would be the way to go. Or use 
a separate library that already provides this functionality.

The native metadata chunks seem to be defined very loosely: 
http://aifftagging.blogspot.de/
"Native metadata - The AIFF specification defines native metadata chunks - 
Name, Author, Comment, Annotation, and Copyright. However, software support 
for these chunks is uncommon."

I suppose there is a reason why they didn't implement this seemingly obvious 
feature. So why should we!?

**Comment by uklotzde** [comment]
Ooops, accidentally deleted my branch ;)

**Comment by daschuer** [comment]
I have investigates this a bit more, because it feels odd to be able to write these tags, but not be able to read them back. 

The required protected function are part of the documented API: 
http://taglib.org/api/classTagLib_1_1RIFF_1_1File.html

So it should be finally not an issue to inherit from TagLib::RIFF::AIFF::File and expose functions for reading the artist and title. This will not required to alter the taglib code, we do this often when inherit from Qt objects. 

The tags itself are documented here: 
https://www2.ak.tu-berlin.de/~fhein/Alias/Studio/ProTools/audio-formate/aiff/overview.html

On the other hand we have no user complaining about it ... :-/
What do you think?

**Comment by uklotzde** [comment]
I didn't want to implement that, but here you have it.

**Comment by daschuer** [comment]
Thank you, and sorry for putting extra work on you.

**Comment by daschuer** [comment]
It works like a charm. And LGTM!
Thanks.

File: src/engine/sidechain/enginerecord.cpp [hunk 1]
```diff
@@ -363,20 +363,31 @@ bool EngineRecord::openFile() {
...
+                    qWarning("libsndfile error: %s", sf_error_number(ret));
+                }
             }
 
-            ret = sf_set_string(m_pSndfile, SF_STR_ALBUM, m_baAlbum.constData());
-            if (ret != 0) {
-                qWarning("libsndfile error: %s", sf_error_number(ret));
+            if (!m_baAlbum.isEmpty()) {
+                int strType = SF_STR_ALBUM;
+                if (ENCODING_AIFF) {
```
Comment (JosepMaJAZ) [inline src/engine/sidechain/enginerecord.cpp]: Btw.. here there's a bug ;) But if my wavesettings PR gets merged soon, you don't need to fix it.

**Comment by uklotzde** [comment]
Oops. I finally only tested the AIFF case and missed to re-test WAV! Thanks 
for keeping an eye on it :)

**Pull request merged**